pysph/tools/interpolator.py

```diff
@@ -285,10 +420,11 @@ def _create_particle_array(self, x, y, z):
             x=xr, y=yr, z=zr, h=h,
             number_density=np.zeros_like(xr),
             prop=prop,
-            grad_x=np.zeros_like(xr),
-            grad_y=np.zeros_like(xr),
-            grad_z=np.zeros_like(xr)
         )
+        if not self.use_shepard:
+            pa.add_property('A', stride=16)
```

Comment:
These names are rather common, for example if the input array had a property called 'A' this would not work.  Can you use a more unique name and something a bit more meaningful.  Same for B and `propxyz`?

Reply:
done